File: src/main/java/frc/team4069/robot/commands/SimpleDriveCommand.java
package frc.team4069.robot.commands;

// An example command for driving straight for 3 meters
public class SimpleDriveCommand extends CommandBase {

    // Called to initialize the drive base
    protected void initialize() {
        // Claim exclusive use of the drive base
        requires(driveBase);
        // Start driving straight at 50% speed
        driveBase.driveContinuousSpeed(0, 0.5);
    }

    // Called to check whether this command has completed
    protected boolean isFinished() {
        // The command has finished if and only if the drive base has traveled 3 meters
        return driveBase.getDistanceTraveledMeters() >= 3;
    }

    // Called when the command is exiting
    protected void end() {
        // Stop driving and coast to a halt
        driveBase.stop();
    }
}

File: src/main/java/frc/team4069/robot/commands/SimpleDriveCommand.kt
package frc.team4069.robot.commands

import edu.wpi.first.wpilibj.command.Command
import frc.team4069.robot.Robot

// An example command for driving straight five meters
class SimpleDriveCommand : Command() {

    // Called to initialize the drive base
    override fun initialize() {
        // Claim exclusive use of the drive base
        requires(Robot.driveBase)
        // Start driving straight at 50% speed
        Robot.driveBase.driveContinuousSpeed(0.0, 0.5)
    }

    // Called to check whether this command has completed
    override fun isFinished(): Boolean = Robot.driveBase.distanceTraveledMeters >= 5
}
